src/configuration/configuration.py

```diff
@@ -0,0 +1,61 @@
+# configuration_manager.py
+# author: Josue Mendoza
+# date: 4-2-2015
+
+from xml.dom.minidom import *
+from file_manager import File
+from singleton import Singleton
+
+class Configuration(object):
+
+	__metaclass__ = Singleton
+
+	CONFIGURATION_NAME = 'configuration'
+	LEVEL_NAME = 'level'
+	ALGORITHM_NAME = 'algorithm'
+	FILEPATH_SAVE_NAME = 'filepath_save'
+	FILENAME_SAVE_NAME = 'filename_save'
+
+	def __init__(self, xml_content = None):
+		self.__raw_xml_configuration = xml_content
+		self.level = self.get_value_from_raw_xml(self.LEVEL_NAME)
+		self.algorithm = self.get_value_from_raw_xml(self.ALGORITHM_NAME)
+		self.filepath_save = self.get_value_from_raw_xml(self.FILEPATH_SAVE_NAME)
+		self.filename_save = self.get_value_from_raw_xml(self.FILENAME_SAVE_NAME)
+
+	def get_value_from_raw_xml(self, xml_key):
+		"""Gets a value from a XML string.
```

Comment:
- What is the return value of this method, that should be on the documentation too.
- What are the possible values or at least provide an examples of the possible values